src/components/PostDetails.tsx

```diff
@@ -1,106 +1,114 @@
 import React from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { deleteComment } from '../api/comments';
+
+type Props = {
+  post: Post | null;
+  comments: Comment[];
+  setComments: (comments: Comment[]) => void;
+  loadingComments: boolean;
+  errorLoadComments: boolean;
+  writeComment: boolean;
+  setWriteComment: (writeCommen: boolean) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  comments,
+  setComments,
+  loadingComments,
+  errorLoadComments,
+  writeComment,
+  setWriteComment,
+}) => {
+  function removeComment(comment: Comment) {
+    setComments([...comments.filter(commentary => commentary !== comment)]);
+    deleteComment(comment.id).catch(() => setComments([...comments]));
```

Comment:
Consider adding a notification or message to inform the user if the comment deletion fails. This will improve user experience by providing feedback on the action.